hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDataBlock.java

```diff
@@ -115,6 +114,35 @@ public byte[] getContentBytes() throws IOException {
     return serializeRecords(records.get());
   }
 
+  private Schema getReaderSchema(Option<Schema> readerSchemaOpt) {
+    Schema writerSchema = getWriterSchema(super.getLogBlockHeader());
+    // If no reader-schema has been provided assume writer-schema as one
+    if (!readerSchemaOpt.isPresent()) {
```

Comment:
Could you clarify when the reader schema is setup?

Reply:
It's setup here: `org.apache.hudi.HoodieBaseRelation#buildScan`, where the it is usually derived from `org.apache.hudi.HoodieBaseRelation#tableAvroSchema`